unitylibs/core/workflow/workflow-acrobat/action-binder.js

```diff
@@ -264,14 +275,58 @@ export default class ActionBinder {
     return fileTypes.size > 1 ? 'mixed' : files[0].type;
   }
 
+  async sanitizeFileName(rawFileName) {
+    try {
+      const { getExtension, withoutExtension } = await import('../../../utils/StringUtils.js');
+
+      const MAX_FILE_NAME_LENGTH = 255;
+      let fileName = rawFileName;
+
+      if (!fileName || fileName === '.' || fileName === '..') {
+        return '---';
+      }
+      
+      let ext = getExtension(fileName);
+      const nameWithoutExtension = withoutExtension(fileName);
+
+      ext = ext.length > 0 ? `.${ext}` : '';
+
+      fileName = DOS_SPECIAL_NAMES.has(nameWithoutExtension.toUpperCase()) 
+        ? `---${ext}` 
+        : nameWithoutExtension + ext;
+
+      if (fileName.length > MAX_FILE_NAME_LENGTH) {
+        const trimToLen = MAX_FILE_NAME_LENGTH - ext.length;
+        fileName = trimToLen > 0 ? fileName.substring(0, trimToLen) + ext : fileName.substring(0, MAX_FILE_NAME_LENGTH);
+      }
+
+      fileName = fileName
+        .replace(ENDING_SPACE_PERIOD_REGEX, '-')
+        .replace(STARTING_SPACE_PERIOD_REGEX, '-')
+        .replace(INVALID_CHARS_REGEX, '-');
+
+      if (rawFileName !== fileName) {
+        await this.dispatchErrorToast('verb_warn_renamed_invalid_file_name', null, `Renamed ${rawFileName} to ${fileName}`, true)
+      }
+      return fileName;
+    } catch (error) {
+      console.error('Error sanitizing filename:', error);
+      await this.dispatchErrorToast('verb_upload_error_generic', 500, `Error renaming file: ${rawFileName}`, false);
+      return '---';
+    }
+  }
```

Comment:
Please avoid the new line spacing between statements of code within each function if not needed. This is typically avoided in the milo world as it can be beneficial for performance.